src/js/express.js

```diff
@@ -0,0 +1,148 @@
+// Express instance represented as a component
+//
+// Add additional routers and middleware as child components of this module
+"use strict";
+var fluid       = fluid || require("infusion");
+var namespace   = "gpii.express";
+var express     = fluid.registerNamespace(namespace);
+var path        = require("path");
+var exphbs      = require("express-handlebars");
+var handlebars  = require("handlebars");
+
+express.initExpress = function(that) {
+    if (!that.options.config || !that.options.config.express) {
+        console.error("Cannot initialize express because you have not supplied a 'config' option.");
+        return;
+    }
+
+    // We need these to be model variables so that other grades can listen and take action when they are changed.
+    var express  = require("express");
+    that.applier.change("express", express());
+    that.applier.change("router", express.Router());
+    that.model.express.use(that.options.path, that.model.router);
+
+    var helpersToLoad = [];
+    if (that.options.components) {
```

Comment:
All this polymorphic code is wrapped up in a function and is not accessible for people extending the system to customise. Also, looking inside ones options for component definitions is not a reliable way to react to all sources by which subcomponents may be defined. 

Instead, you should write distributeOptions definitions targetting each of the kinds of components which you recognise, distributing down event handlers which they can fire on creation.

For example

distibuteOptions: {
    record: "{gpii.express}.registerRouter",
   target: "{that gpii.express.router}.listeners.onCreate"
   }

Then the body, for example, of your first "if" statement block can be written inside a method (invoker) called "registerRouter" will which be automatically called on startup of any matching subcomponent.

This system can then be gracefully extended by both you and other extenders of your base grade by writing further distribution blocks and further invokers in derived grades of the base.